linera-service/src/node_service.rs

```diff
@@ -141,8 +188,14 @@ where
     ViewError: From<S::ContextError>,
 {
     /// Subscribes to notifications from the current chain.
-    async fn notifications(&self) -> Result<impl Stream<Item = Notification>, Error> {
-        Ok(ChainClient::listen(self.client.clone()).await?)
+    async fn notifications(
+        &self,
+        chain_id: ChainId,
+    ) -> Result<impl Stream<Item = Notification>, Error> {
+        let Some(client) = self.clients.client(&chain_id).await else {
```

Comment:
TBH I like the `let Some(client) =` because it uses only three lines, and it's used in lots of places. The alternative would be `rustfmt`ted to use many more lines, some of which are just `.await`. I find that more visually noisy, but it's a matter of taste.